lab/src/Microsoft.SyndicationFeed/src/RssWriter/Rss20Formatter2.cs

```diff
@@ -0,0 +1,406 @@
+
+        private ISyndicationContent CreateLinkContent(ISyndicationLink link)
+        {
+            //Write attributes if exist
+            var content = new SyndicationContent(link.RelationshipType);
+
+            //
+            // Lenght
+            if (link.Length != 0)
+            {
+                content.AddAttribute(new SyndicationAttribute(Rss20Constants.LengthTag, FormatValue(link.Length)));
+            }
+
+            //
+            // Type
+            if (!string.IsNullOrEmpty(link.MediaType))
+            {
+                content.AddAttribute(new SyndicationAttribute(Rss20Constants.TypeTag, link.MediaType));
+            }
+
+            //
+            // url attribute
+            if (link.Uri.OriginalString != link.Title)
+            {
+                if (string.IsNullOrEmpty(link.Title))
+                {
+                    content.Value = FormatValue(link.Uri);
+                }
+                else
+                {
+                    content.AddAttribute(new SyndicationAttribute(Rss20Constants.UrlTag, FormatValue(link.Uri)));
+                }
+            }
+
+            //
+            // title 
+            if (!string.IsNullOrEmpty(link.Title))
+            {
+                content.Value = link.Title;
+            }
+
+
+            return content;
+        }
+
+        private ISyndicationContent CreateCommentsContent(ISyndicationLink link)
+        {
+            return new SyndicationContent(link.RelationshipType)
+            {
+                Value = FormatValue(link.Uri)
+            };
+        }
+
+        private ISyndicationContent CreateSourceContent(ISyndicationLink link)
+        {
+            var content = new SyndicationContent(link.RelationshipType);
+
+            //
+            // Url
+            content.AddAttribute(new SyndicationAttribute(Rss20Constants.UrlTag, FormatValue(link.Uri)));
+
+            //
+            // Title
+            if (!string.IsNullOrEmpty(link.Title))
+            {
+                content.Value = link.Title;
+            }
+
+            return content;
+        }
+        
+        protected virtual ISyndicationContent CreateContent(ISyndicationLink link)
+        {
+            if (link == null)
+            {
+                throw new ArgumentNullException(nameof(link));
+            }
+
+            switch (link.RelationshipType)
+            {
+                case Rss20Constants.LinkTag:
+                    return CreateLinkContent(link);
+
+                case Rss20Constants.EnclosureTag:
+                    return CreateEnclosureContent(link);
+
+                case Rss20Constants.CommentsTag:
+                    return CreateCommentsContent(link);
+
+                case Rss20Constants.SourceTag:
+                    return CreateSourceContent(link);
+
+                default:
+                    return null;
+            }
+        }
+
+        protected virtual void Write(ISyndicationContent content, XmlWriter writer)
```

Comment:
Why protected? The extensibility is based on overriding CreateContent(...) methods, right?